examples/vision/semisupervised_simclr.py

```diff
@@ -0,0 +1,658 @@
+            RandomColorAffine(brightness, jitter),
+        ]
+    )
+
+
+def visualize_augmentations(num_images):
+    # sample a batch from a dataset
+    images = next(iter(train_dataset))[0][0][:num_images]
+    # apply augmentations
+    augmented_images = zip(
+        images,
+        get_augmenter(**classification_augmentation)(images),
+        get_augmenter(**contrastive_augmentation)(images),
+        get_augmenter(**contrastive_augmentation)(images),
+    )
+
+    row_titles = [
+        "Original:",
+        "Weakly augmented:",
+        "Strongly augmented:",
+        "Strongly augmented:",
+    ]
+    plt.figure(figsize=(num_images * 2.2, 4 * 2.2), dpi=100)
+    for column, image_row in enumerate(augmented_images):
+        for row, image in enumerate(image_row):
+            plt.subplot(4, num_images, row * num_images + column + 1)
+            plt.imshow(image)
+            if column == 0:
+                plt.title(row_titles[row], loc="left")
+            plt.axis("off")
+    plt.tight_layout()
+
+
+visualize_augmentations(num_images=8)
+
+"""
+## Encoder architecture
+"""
+
+
+# define the encoder architecture
+def get_encoder():
+    return keras.Sequential(
+        [
+            layers.Input(shape=(image_size, image_size, image_channels)),
+            layers.Conv2D(width, kernel_size=3, strides=2, activation="relu"),
+            layers.Conv2D(width, kernel_size=3, strides=2, activation="relu"),
+            layers.Conv2D(width, kernel_size=3, strides=2, activation="relu"),
+            layers.Conv2D(width, kernel_size=3, strides=2, activation="relu"),
+            layers.Flatten(),
+            layers.Dense(width, activation="relu"),
+        ],
+        name="encoder",
+    )
+
+
+"""
+## Supervised baseline model
+
+A baseline supervised model is trained using random initialization for 60
+epochs.
+"""
+
+# baseline supervised training with random initialization
+baseline_model = keras.Sequential(
+    [
+        layers.Input(shape=(image_size, image_size, image_channels)),
+        get_augmenter(**classification_augmentation),
+        get_encoder(),
+        layers.Dense(10),
+    ],
+    name="baseline_model",
+)
+baseline_model.compile(
+    optimizer=keras.optimizers.Adam(),
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    metrics=[keras.metrics.SparseCategoricalAccuracy(name="acc")],
+)
+
+baseline_history = baseline_model.fit(
+    labeled_train_dataset, epochs=num_epochs, validation_data=test_dataset
+)
+print(
+    "Maximal validation accuracy: {:.2f}%".format(
+        max(baseline_history.history["val_acc"]) * 100
+    )
+)
+
+"""
+## Self-supervised model for contrastive pretraining
+
+We pretrain an encoder on unlabeled images with a contrastive loss for 30
+epochs. A nonlinear projection head is attached to the top of the encoder, as it
+improves the quality of representations of the encoder.
+
+The InfoNCE/NT-Xent/N-pairs loss is used, which can be interpreted in the
```

Comment:
Suggestion: use active voice

```suggestion
We use the InfoNCE/NT-Xent/N-pairs loss, which can be interpreted in the
```